hibernate-core/src/main/java/org/hibernate/internal/SessionFactoryImpl.java

```diff
@@ -260,6 +261,11 @@ public SessionFactoryImpl(
 			//noinspection unchecked
 			tenantIdentifierJavaType = jdbcMapping.getJavaTypeDescriptor();
 		}
+		for (Map.Entry<String, FilterDefinition> filterEntry : filters.entrySet()) {
+			if (filterEntry.getValue().isAutoEnabled()) {
+				autoEnabledFilters.add( filterEntry.getValue() );
+			}
+		}
```

Comment:
```suggestion
		autoEnabledFilters = new HashSet<>( filters.values() );
```

Reply:
Shouldn't `autoEnabledFilters` contains only the `FilterDefinition`s that have the autoEnabled flag set to true? With this suggestion, we wouldn't filter them anymore